src/SourceGenerators/Uno.UI.SourceGenerators.Tests/XamlCodeGeneratorTests/Out/TTLNMDWSRD/XamlCodeGenerator_MyResourceDictionary_92716e07ff456818f6d4125e055d4d57.cs

```diff
@@ -0,0 +1,112 @@
+#elif __MACOS__
+using _View = AppKit.NSView;
+#elif UNO_REFERENCE_API || IS_UNIT_TESTS
+using _View = Windows.UI.Xaml.UIElement;
+#endif
+
+namespace MyProject
+{
+	public sealed partial class GlobalStaticResources
+	{
+		[global::System.ComponentModel.EditorBrowsable(global::System.ComponentModel.EditorBrowsableState.Never)]
+		private const string __baseUri_prefix_MyResourceDictionary_92716e07ff456818f6d4125e055d4d57 = "ms-appx:///TestProject/";
+		[global::System.ComponentModel.EditorBrowsable(global::System.ComponentModel.EditorBrowsableState.Never)]
+		private const string __baseUri_MyResourceDictionary_92716e07ff456818f6d4125e055d4d57 = "ms-appx:///TestProject/";
+		// This non-static inner class is a means of reducing size of AOT compilations by avoiding many accesses to static members from a static callsite, which adds costly class initializer checks each time.
+		internal sealed class ResourceDictionarySingleton__MyResourceDictionary_92716e07ff456818f6d4125e055d4d57 : global::Uno.UI.IXamlResourceDictionaryProvider
+		{
+			private static global::Windows.UI.Xaml.NameScope __nameScope = new global::Windows.UI.Xaml.NameScope();
+			private static ResourceDictionarySingleton__MyResourceDictionary_92716e07ff456818f6d4125e055d4d57 __that;
+			internal static ResourceDictionarySingleton__MyResourceDictionary_92716e07ff456818f6d4125e055d4d57 Instance
+			{
+				get
+				{
+					if (__that == null)
+					{
+						__that = new ResourceDictionarySingleton__MyResourceDictionary_92716e07ff456818f6d4125e055d4d57();
+					}
+
+					return __that;
+				}
+			}
+
+			internal global::Uno.UI.Xaml.XamlParseContext __ParseContext_ {get; }
+
+			private ResourceDictionarySingleton__MyResourceDictionary_92716e07ff456818f6d4125e055d4d57()
+			{
+				__ParseContext_ = global::MyProject.GlobalStaticResources.__ParseContext_;
+			}
+
+			private global::Windows.UI.Xaml.ResourceDictionary _MyResourceDictionary_92716e07ff456818f6d4125e055d4d57_ResourceDictionary;
+
+			internal global::Windows.UI.Xaml.ResourceDictionary MyResourceDictionary_92716e07ff456818f6d4125e055d4d57_ResourceDictionary
+			{
+				get
+				{
+					if (_MyResourceDictionary_92716e07ff456818f6d4125e055d4d57_ResourceDictionary == null)
+					{
+						_MyResourceDictionary_92716e07ff456818f6d4125e055d4d57_ResourceDictionary = 
+						new global::Windows.UI.Xaml.ResourceDictionary
+						{
+							IsParsing = true,
+							MergedDictionaries = {
+							// Source not resolved statically, falling back on external resource retrieval.
+							global::Uno.UI.ResourceResolver.RetrieveDictionaryForSource("ms-appx:///Path/To/File1.xaml", "C:/Project/0/MyResourceDictionary.xaml")
+							.AddMergedDictionaries(
+							// MergedDictionaries
+							// Source not resolved statically, falling back on external resource retrieval.
+							global::Uno.UI.ResourceResolver.RetrieveDictionaryForSource("ms-appx:///Path/To/File2.xaml", "C:/Project/0/MyResourceDictionary.xaml")
+							)
```

Comment:
Due to the syntactic position of the merged dictionary (being in collection initializer), it was hard to adjust for the nested dictionaries without the new `AddMergedDictionaries` extension method.

@kazo0 Let me know if this new code gen makes sense to you.

Reply:
@jeromelaban I think we might want to consider moving away from collection initializers. While they are very nice syntax, they can sometimes complicate things for source generation and forces us to add new extension methods that modifies some instance state and then return the instance, and also having us to have flags in the source generator code to tell when we are in different syntactic contexts to use appropriate language punctuation for the specific context.